src/ds5/ds5-factory.cpp

```diff
@@ -37,6 +37,25 @@ namespace librealsense
               ds5_advanced_mode_base(ds5_device::_hw_monitor, get_depth_sensor()) {}
 
         std::shared_ptr<matcher> create_matcher(const frame_holder& frame) const override;
+        
+        std::vector<tagged_profile> get_profiles_tags() const override
```

Comment:
Suggestion, to consider:
Profile tags are queries every time you call `tag_profile`, so the resulting vector will be created and re-created over and over again. This is only during start-up, but still. 
Perhaps you can add another (non-virtual) getter that would fetch a `lazy<std::vector<tagged_profile>>` that would be initialized with lambda to `get_profiles_tags`?